### PR TITLE
[6.0] affinity antonym

### PR DESCRIPTION
**Pull request by Quezler**
> pardon the switch statement, but that was the most compact thing on my mind right now 😗
 
Screen Shot 2020-03-09 at 13 34 55

> some space between the lines might be necessary, they nearly touch one another's edges 🤔

### CODE REVIEW

**Comment by fuzzbuck** [comment]
perhaps rename them to "Efficient on" & "Inefficient on"? Affinities & Aversions might be hard to understand for players who are not fluent in english

**Comment by fuzzbuck** [comment]
Otherwise nice pr, makes sense that it wont work at full speed on literal lava lol

**Comment by Quezler** [comment]
I'll leave the naming of them up to anuke, i just went with this word since it mirrors the existing one, and the java does look appealing but it could indeed use some improvements, any ideas? 🤔

**Comment by Quezler** [comment]
as discussed in the #1093 pull "more/less efficient on" would take up space, here's a concept:
Screen Shot 2020-03-10 at 12 25 14
> obviously this will require some invasive code edits to get right, bundles/stats and such 😗

File: core/src/mindustry/world/meta/BlockStats.java [hunk 1]
```diff
@@ -38,8 +38,16 @@ public void add(BlockStat stat, Liquid liquid, float amount, boolean perSecond){
     }
 
     public void add(BlockStat stat, Attribute attr){
+        add(stat, attr, +1);
+    }
+
+    public void add(BlockStat stat, Attribute attr, int sign){
         for(Block block : Vars.content.blocks()){
-            if(!block.isFloor() || block.asFloor().attributes.get(attr) <= 0) continue;
```
Comment (Anuken) [inline core/src/mindustry/world/meta/BlockStats.java]: The attribute sign check can be written (without the switch) as `if(!block.isFloor() || !Mathf.equal(Math.signum(block.asFloor().attributes.get(attr)), sign)) continue;`

**Comment by Anuken** [comment]
Another option for naming is to ditch the concept of affinities/aversions completely and display a `Efficiencies:` list with each tile having a % number on it, possibly colored in red/green to help indicate whether it's better or worse.

**Comment by Quezler** [comment]
> Another option for naming is to ditch the concept of affinities/aversions completely and display a `Efficiencies:` list with each tile having a % number on it, possibly colored in red/green to help indicate whether it's better or worse.

how would that translate to the amount of tiles the (multi)block is on? could get confusing 🤔

**Comment by Anuken** [comment]
> could get confusing

Not at all. +20% could simply mean "20% more efficient for every block this producer is on", and -20% would mean the reverse. Being on two 20% blocks and having everything else be normal would lead to 140% efficiency.

**Pull request closed without merging**